### PR TITLE
[docs] add Doxygen support for CLI comments

### PR DESCRIPTION
**Pull request by sarah-iot**

### CODE REVIEW

File: src/cli/cli_joiner.cpp [hunk 1]
```diff
@@ -86,6 +127,20 @@ template <> otError Joiner::Process<Cmd("id")>(Arg aArgs[])
...
 }
 
+/**
+ * @cli joiner start
+ * @code
+ * joiner start J01NM3
+ * Done
+ * @endcode
+ * @cparam joiner start @ca{joining-device-credential} [@ca{provisioning-url}]
+ * *   `joining-device-credential`: %Joiner Passphrase. Must be a string of all uppercase alphanumeric
```
Comment (Vyrastas) [inline src/cli/cli_joiner.cpp:137]: What is the % used for here?  I see it prefixing Joiner in multiple places.  It's not mentioned in the contribute page

**Review by Vyrastas — APPROVED** [review]
Great job on this, LGTM from a TW perspective

File: src/cli/cli_joiner.cpp (hunk 1 above)
Comment (sarah-iot) [reply src/cli/cli_joiner.cpp:137]: I updated the Contributing page and added some instructions for the `%`. It's used to tell Doxygen to NOT link to a word...it kept thinking that Joiner was a class name...

**Review by abtink — APPROVED** [review]
LGTM. Thanks @sarah-iot. 👍

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**